### PR TITLE
In band registration example

### PR DESCRIPTION
**Pull request by GavinDmello**

### CODE REVIEW

File: examples/in-band-registeration.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+});
+
+//response stanzas
+client.on('stanza', function (stanza) {
+ if (stanza.attrs.type == 'error') {
+        console.log('[error] ' + stanza);
+        return;
+    }
+
+    /*
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: Could you mark this as specific to ejabberd please?

File: examples/in-band-registeration.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+       .c('query',{ xmlns: 'jabber:iq:register'})
+        .c('username').t('user').up()  // Give a username
+        .c('password').t('1234')  //Give a password
+    client.send(stanza) //stanza
+
+});
+
+//response stanzas
+client.on('stanza', function (stanza) {
+ if (stanza.attrs.type == 'error') {
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: '==='

File: examples/in-band-registeration.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  console.error(error)
+  process.exit(1)
+});
+
+
+client.on('online', function (data) {
+    //stanza to be sent to the server
+    var stanza = new Client.Stanza('iq', {type:'set',id:'reg1',to:'some.server.org'})
+       .c('query',{ xmlns: 'jabber:iq:register'})
+        .c('username').t('user').up()  // Give a username
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: indentation issues?

File: examples/in-band-registeration.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+//check for errors on the socket
+client.connection.socket.on('error', function (error) {
+  console.error(error)
+  process.exit(1)
+});
+
+
+client.on('online', function (data) {
+    //stanza to be sent to the server
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: space after '//'

File: examples/in-band-registeration.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+'use strict'
+var Client = require('node-xmpp-client')
+
+//connect using the admin user's creds
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: space after '//'

File: examples/in-band-registeration.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+var client = new Client({jid: 'admin@some.server.org', password: '1234'})
+
+//check for errors on the socket
+client.connection.socket.on('error', function (error) {
+  console.error(error)
+  process.exit(1)
+});
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: remove ';'

File: examples/in-band-registeration.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+
+client.on('online', function (data) {
+    //stanza to be sent to the server
+    var stanza = new Client.Stanza('iq', {type:'set',id:'reg1',to:'some.server.org'})
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: { type: 'set', id: 'reg1', to: 'some.server.org' }  /* formatting */

File: examples/in-band-registeration.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+       .c('query',{ xmlns: 'jabber:iq:register'})
+        .c('username').t('user').up()  // Give a username
+        .c('password').t('1234')  //Give a password
+    client.send(stanza) //stanza
+
+});
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: remove ';'

File: examples/in-band-registeration.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+//response stanzas
+client.on('stanza', function (stanza) {
+ if (stanza.attrs.type == 'error') {
+        console.log('[error] ' + stanza);
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: could you consistently use 2 spaces for indentation

File: examples/in-band-registeration.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    change {access, register} to {access_from, register}
+
+    and
+
+    comment out the ip_access section in your ejabberdctl.cfg
+
+
+    */
+
+});
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: remove ';'

File: examples/in-band-registeration.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+client.on('error', function (err) {
+  console.error(err)
+  process.exit(1)
+});
```
Comment (lloydwatkin) [inline examples/in-band-registeration.js]: remove ';'

**Comment by lloydwatkin** [comment]
Thanks very much for your CR. I've included a few comments to align your example with our code formatting rules and then we'll be good to merge in :)

File: examples/in-band-registeration.js (hunk 1 above)
Comment (GavinDmello) [inline examples/in-band-registeration.js]: How can I mark it specific to ejabberd?

**Comment by GavinDmello** [comment]
I've made the changes. Have a look if possible :)

**Comment by lloydwatkin** [comment]
Thanks!

**Comment by sonnyp** [comment]
@GavinDmello thx

**Comment by sonnyp** [comment]
@lloydwatkin tests (linter) is failing for this PR let's make sure only :green_apple: PRs are merged

**Comment by GavinDmello** [comment]
I guess its because I did a require('node-xmpp-client') instead of requiring the index.js file in the first place.

**Comment by lloydwatkin** [comment]
There were a couple of small tweaks and I merged with updates rather than have a back-and-forth, had there been more I would have let you know. Thanks for the contribution :)

**Comment by GavinDmello** [comment]
Cool :)

**Pull request merged**